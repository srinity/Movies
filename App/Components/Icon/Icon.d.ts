import { Component } from 'react'
import { TextProps, StyleProp, TextStyle } from 'react-native'

export type IconTypes =
  | 'material'
  | 'ant'
  | 'entypo'
  | 'evil'
  | 'feather'
  | 'fontAwesome'
  | 'fontisto'
  | 'foundation'
  | 'ion'
  | 'materialCommunity'
  | 'oct'
  | 'simpleLine'
  | 'zocial'

export type IconName = string

export interface IconProps extends TextProps {
  /**
   * The name of the icon in the specified type
   *
   */
  name: IconName

  /**
   * The type of the icon used
   *
   * @default 'material'
   */
  type?: IconTypes

  /**
   * The size of the icon used
   *
   * @default 25
   */
  size?: number

  /**
   * The color of the icon used
   *
   * @default '#000000'
   */
  color?: string

  /**
   * optional styling applied to the icon
   *
   * @default {}
   */
  style?: StyleProp<TextStyle>
}

export default class Icon extends Component<IconProps> {}
