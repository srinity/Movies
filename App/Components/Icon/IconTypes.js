import { values as _values } from 'lodash'

const iconTypes = {
  material: 'material',
  ant: 'ant',
  entypo: 'entypo',
  evil: 'evil',
  feather: 'feather',
  fontAwesome: 'fontAwesome',
  fontisto: 'fontisto',
  foundation: 'foundation',
  ion: 'ion',
  materialCommunity: 'materialCommunity',
  oct: 'oct',
  simpleLine: 'simpleLine',
  zocial: 'zocial'
}

export const iconTypesValues = _values(iconTypes)

export default iconTypes
