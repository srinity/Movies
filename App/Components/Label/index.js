import Label, { AnimatedLabel } from './Label'

export { AnimatedLabel }
export default Label
