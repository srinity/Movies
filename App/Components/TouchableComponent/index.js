import TouchableComponent from './TouchableComponent'

export default TouchableComponent
