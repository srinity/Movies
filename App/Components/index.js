import Header from './Header'
import { Icon, IconTypes } from './Icon'
import Screen from './Screen'

export { Header, Icon, IconTypes, Screen }
