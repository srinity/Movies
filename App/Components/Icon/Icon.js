import React, { useMemo } from 'react'
import { ViewPropTypes } from 'react-native'
import PropTypes from 'prop-types'
import AntIcon from 'react-native-vector-icons/AntDesign'
import EntypoIcon from 'react-native-vector-icons/Entypo'
import EvilIcon from 'react-native-vector-icons/EvilIcons'
import FeatherIcon from 'react-native-vector-icons/Feather'
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome'
import Fontisto from 'react-native-vector-icons/Fontisto'
import FoundationIcon from 'react-native-vector-icons/Foundation'
import Ionicon from 'react-native-vector-icons/Ionicons'
import MaterialCommunityIcon from 'react-native-vector-icons/MaterialCommunityIcons'
import MaterialIcon from 'react-native-vector-icons/MaterialIcons'
import OctIcon from 'react-native-vector-icons/Octicons'
import SimpleLineIcon from 'react-native-vector-icons/SimpleLineIcons'
import ZocialIcon from 'react-native-vector-icons/Zocial'

import { colorPropType } from '../../Utils/PropTypesValidators'

import IconTypes, { iconTypesValues } from './IconTypes'

const determineProperIconTag = iconType => {
  switch (iconType) {
    case IconTypes.material:
      return MaterialIcon
    case IconTypes.ant:
      return AntIcon
    case IconTypes.entypo:
      return EntypoIcon
    case IconTypes.evil:
      return EvilIcon
    case IconTypes.feather:
      return FeatherIcon
    case IconTypes.fontAwesome:
      return FontAwesomeIcon
    case IconTypes.fontisto:
      return Fontisto
    case IconTypes.foundation:
      return FoundationIcon
    case IconTypes.ion:
      return Ionicon
    case IconTypes.materialCommunity:
      return MaterialCommunityIcon
    case IconTypes.oct:
      return OctIcon
    case IconTypes.simpleLine:
      return SimpleLineIcon
    case IconTypes.zocial:
      return ZocialIcon
    default:
      return MaterialIcon
  }
}

const Icon = ({ type, name, color, size, style, ...props }) => {
  const IconComponent = useMemo(() => determineProperIconTag(type), [type])

  return (
    <IconComponent
      name={name}
      color={color}
      size={size}
      style={style}
      {...props}
    />
  )
}

Icon.defaultProps = {
  type: IconTypes.material,
  color: '#000000',
  size: 25,
  style: {}
}

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf([...iconTypesValues]),
  size: PropTypes.number,
  color: colorPropType,
  style: ViewPropTypes.style
}

export default Icon
