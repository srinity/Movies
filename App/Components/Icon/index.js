import Icon from './Icon'
import IconTypes, { iconTypesValues } from './IconTypes'

export { Icon, IconTypes, iconTypesValues }
