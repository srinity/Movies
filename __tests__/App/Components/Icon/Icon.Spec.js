import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { Icon, IconTypes } from '../../../../App/Components/Icon'

describe('Icon Component', () => {
  it('should render icon with default type correctly', () => {
    const icon = renderer.create(<Icon name="add" />)
    expect(icon).toMatchSnapshot()
  })

  it('should render icon of type antIcon correctly', () => {
    const icon = renderer.create(<Icon type={IconTypes.ant} name="forward" />)
    expect(icon).toMatchSnapshot()
  })

  it('should be able to change icon color', () => {
    const props = {
      type: IconTypes.ant,
      name: 'forward',
      color: '#888888'
    }

    const icon = shallow(<Icon {...props} />)
    expect(icon.prop('color')).toEqual(props.color)
  })

  it('should be able to change the icon size', () => {
    const props = {
      name: 'add',
      size: 50
    }

    const icon = shallow(<Icon {...props} />)
    expect(icon.prop('size')).toEqual(props.size)
  })

  it('should be able to add custom valid style', () => {
    const props = {
      name: 'add',
      style: {
        elevation: 4,
        opacity: 0.8,
        marginVertical: 10
      }
    }

    const icon = shallow(<Icon {...props} />)
    expect(icon.prop('style')).toEqual(props.style)
  })
})
