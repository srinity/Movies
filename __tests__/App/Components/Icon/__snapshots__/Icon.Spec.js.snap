// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component should render icon of type antIcon correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 25,
      },
      Object {},
      Object {
        "fontFamily": "anticon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Icon Component should render icon with default type correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 25,
      },
      Object {},
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;
